exonum-java-binding/core/rust/src/utils/mod.rs

```diff
@@ -27,3 +27,20 @@ pub use self::errors::{
     panic_on_exception, unwrap_exc_or, unwrap_exc_or_default, unwrap_jni, unwrap_jni_verbose,
 };
 pub use self::jni::{get_class_name, get_exception_message};
+
+/// Asserts that given closure panics while executed and the resulting error message contains given
+/// substring.
+#[cfg(test)]
+pub fn assert_panics<F, R>(err_substring: &str, f: F)
+where
+    F: FnOnce() -> R,
+{
+    let result = std::panic::catch_unwind(std::panic::AssertUnwindSafe(f));
+    match result {
+        Ok(_) => panic!("Panic expected"),
+        Err(err) => {
+            let err_msg = any_to_string(&err);
+            assert!(err_msg.contains(err_substring));
```

Comment:
Is that correct that it won't show neither a substring nor the actual string? 

I'd suggest including both in the form 
```
Expecting [panic] message ("%s") to contain "%s"
```

Or, as in Java:
```
Expected a panic message containing "bar", but was "foo"
```

Reply:
After some testing, I tend to the second variant because it fits better for multiline actual strings. Also, for better readability I print the actual message from the new line.